### ISSUE TITLE
build/{id}/run throws 404

### ISSUE THREAD

**Issue by rolandoldengarm** [Microsoft/TACO#248]
Set up: Visual Studio 2015, Mac Remotebuild server.
When deploying from VS, everything works fine, except the fact that the app is not started. When looking at the console on the Mac, I see:
    GET /cordova/build/12815/run 404

All other requests return 200.
Can't find anything on the internet about this?

**Comment by MSLaguana** [comment]
If you open up the "output" console in Visual Studio, you should see some extra information. What does it say towards the end there? There are a number of reasons that it could be failing to launch, and the information in the output log should help clarify the cause.

**Comment by rolandoldengarm** [comment]
```
Starting launch process C:\Program Files (x86)\nodejs\node.exe "C:\Users\user\AppData\Roaming\npm\node_modules\vs-tac\remoteDevice.js" --platform ios --action launch --path "C:\Projects\PhoneApp\bld\ios\Debug\buildInfo.json" --serverUrl http://mac:3000/cordova --language en-NZ --loglevel warn  --cliVersion 5.4.0 
Timed out connecting debugger to remote Apache Cordova app. See Output window for JavaScript console output.
Unable to find the app to debug. Check that the device is awake and unlocked.
------ Cordova tools 5.4.0 already installed.
Requesting launch on remote iOS device for buildNumber 12815 on server http://mac:3000/cordova...
Failed to launch iOS remote for build C:\Projects\PhoneApp\bld\ios\Debug\buildInfo.json to http://mac:3000/cordova :
Http 404: Unable to launch app
------ Cordova tools 5.4.0 already installed.
Requesting debug on remote iOS device for buildNumber 12815 on server http://mac:3000/cordova...
Debugging - Successfully Debugging on the device
{webDebugPort=9221}
```
Phone is unlocked & active.

**Comment by MSLaguana** [comment]
That `Unable to launch app` error occurs when there is a failure to communicate with `idevicedebugserverproxy`, and unfortunately we typically don't get much information to work off.

Can you please reproduce the error, and then without turning off remotebuild run `ps ax | grep idevicedebugserverproxy` on the mac and check whether the process is running? If it is, you could try killing it and then running it yourself using the same port that remotebuild would: `idevicedebugserverproxy 3001` and see if that mentions any errors. If that appears to work, try resubmitting your build with idevicedebugserverproxy still running, and see if any errors are printed as part of the communication.

If none of that works, we can try some more detailed debugging.

**Comment by rolandoldengarm** [comment]
Thanks. I get the following error from the proxy:
````
recv failed: Undefined error: 0
Receive failed: Bad file descriptor
````

**Comment by MSLaguana** [comment]
What version of iOS is running on the phone? Around previous iOS releases I had seen some flakiness in launching as sometimes the protocol to talk to the devices changes. Unfortunately it isn't giving us a lot to work with...

You could also try running `brew update && brew upgrade libimobiledevice` to see if there is an updated version which fixes the issue.

**Comment by rolandoldengarm** [comment]
iOS 9.2 is running on the device (iPhone 6 & iPhone 5)

When I run upgrade, it says : 
    limimobiledevice 1.2.0 already installed.

So it seems I already have latest version.

Anything else I can test? It's quite annoying. Debugging only works when I open the app manually at the right time, otherwise I have to try again.

**Comment by MSLaguana** [comment]
The version of Cordova shouldn't matter here, since the step that is failing is launching an application that has already been built, which does not differ based on the contents of the application.

I'm checking to see if I can get it to repro at all on my end. Does your app id (e.g. io.cordova.helloworld) have any unusual characters in it, or contain the substring '.app'?

**Comment by rolandoldengarm** [comment]
You must be kidding me.... 
It did contain ".app", changed that to ".mobile" and now it works.

Awesome :dancer:

But this should be either fixed or documented very clearly I think...

**Comment by MSLaguana** [comment]
I'm afraid that you've discovered the same "wait, really?!" bug that I did a few months back. Fun fact; try launching the same application (with the .app name) from within xcode, and I believe it will also fail. Seems to be a limitation of the platform. For the record `.App` is fine, it's just exactly `.app` that can't appear.

Glad we got that sorted out, and it's definitely a candidate for better documentation. Thanks for reporting this :)
